libobs/obs-view.c

```diff
@@ -139,3 +139,29 @@ void obs_view_render(obs_view_t *view)
 
 	pthread_mutex_unlock(&view->channels_mutex);
 }
+
+obs_source_t *obs_view_get_render_source(struct obs_view *view, uint32_t channel)
+{
+	obs_source_t *source;
+	assert(channel < MAX_CHANNELS);
+
+	if (!view)
+		return NULL;
+	if (channel >= MAX_CHANNELS)
+		return NULL;
+
+	pthread_mutex_lock(&view->channels_mutex);
+	source = view->channels[channel];
+	if (source) {
+		if (source->removed) {
+			obs_source_release(source);
+			view->channels[channel] = NULL;
+			source = NULL;
```

Comment:
`obs_view_render` was the only place this removal logic was happening so to preserve that behavior I added this new function instead of just using `obs_view_get_source`.